tools/make-updatedhelp.ps1

```diff
@@ -0,0 +1,141 @@
+    # no yaml header in our docset will ever be that long
+
+    for ($x = 0; $x -lt 30; $x++) {
+        if ($doc[$x] -eq '---') {
+            if ($start -eq -1) {
+                $start = $x
+            }
+            else {
+                if ($end -eq -1) {
+                    $end = $x + 1
+                    break
+                }
+            }
+        }
+    }
+    if ($end -gt $start) {
+        Write-Output ($doc[$end..$($doc.count)] -join "`r`n")
+    }
+    else {
+        Write-Output ($doc -join "`r`n")
+    }
+}
+
+$Version = $DocSet.Name
+Write-Verbose "Version = $Version"
+
+$VersionFolder = $DocSet.FullName
+Write-Verbose "VersionFolder = $VersionFolder"
+
+# For each of the directories, go through each module folder
+Get-ChildItem $VersionFolder -Directory | ForEach-Object -Process {
+    $ModuleName = $_.Name
+    Write-Verbose "ModuleName = $ModuleName"
+
+    $ModulePath = Join-Path $VersionFolder $ModuleName
+    Write-Verbose "ModulePath = $ModulePath"
+
+    $LandingPage = Join-Path $ModulePath "$ModuleName.md"
+    Write-Verbose "LandingPage = $LandingPage"
+
+    $MamlOutputFolder = Join-Path "$WorkingDirectory\maml" "$Version\$ModuleName"
+    Write-Verbose "MamlOutputFolder = $MamlOutputFolder"
+
+    $CabOutputFolder = Join-Path "$WorkingDirectory\updatablehelp" "$Version\$ModuleName"
+    Write-Verbose "CabOutputFolder = $CabOutputFolder"
+
+    if (-not (Test-Path $MamlOutputFolder)) {
+        New-Item $MamlOutputFolder -ItemType Directory -Force > $null
+    }
+
+    # Process the about topics if any
+    $AboutFolder = Join-Path $ModulePath "About"
+
+    if (Test-Path $AboutFolder) {
+        Write-Verbose "AboutFolder = $AboutFolder"
+        Get-ChildItem "$aboutfolder/about_*.md" | ForEach-Object {
+            $aboutFileFullName = $_.FullName
+            $aboutFileOutputName = "$($_.BaseName).help.txt"
+            $aboutFileOutputFullName = Join-Path $MamlOutputFolder $aboutFileOutputName
+
+            $pandocArgs = @(
+                "--from=gfm",
+                "--to=plain+multiline_tables",
+                "--columns=75",
+                "--output=$aboutFileOutputFullName",
+                "--quiet"
+            )
+
+            Get-ContentWithoutHeader $aboutFileFullName | & $pandocExePath $pandocArgs
+        }
+    }
+
+    try {
+        # For each module, create a single maml help file
+        # Adding warningaction=stop to throw errors for all warnings, erroraction=stop to make them terminating errors
+        New-ExternalHelp -Path $ModulePath -OutputPath $MamlOutputFolder -Force -WarningAction Stop -ErrorAction Stop
+
+        # For each module, create update-help help files (cab and helpinfo.xml files)
+        $cabInfo = New-ExternalHelpCab -CabFilesFolder $MamlOutputFolder -LandingPagePath $LandingPage -OutputFolder $CabOutputFolder
+
+        # Only output the cab fileinfo object
+        if ($cabInfo.Count -eq 8) { $cabInfo[-1].FullName }
```

Comment:
Where does the magic number 8 come from?

Reply:
This is just copied from the existing build script. It how `New-ExternalHelpCab` returns its data. I assume we are checking that it returned the expected amount of data so we can display the interesting piece.